devolving_music/models/song_submission.py

```diff
@@ -1,13 +1,35 @@
+from bisect import bisect_left
+from typing import Iterable
+
+from sqlite3 import IntegrityError
 from django.db import models
-from .event import Event
-from .song import Song
+from django.utils.functional import cached_property
 
+from .song import Song
+from devolving_music.lib.elo_scoring import elo_rating
+from .event import Event
 
 class SongSubmission(models.Model):
     event = models.ForeignKey(Event, on_delete=models.CASCADE)
     song = models.ForeignKey(Song, on_delete=models.CASCADE)
     created_at = models.DateTimeField()
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.energy_score = None
+        self.quality_score = None
+        self.post_peak_score = None
+        self.counted_compares = []
```

Comment:
We might as well make this a set, since the only things we're doing are adding to it and checking membership; checking membership is faster for a set